src/Interpreters/Cache/LRUFileCachePriority.cpp

```diff
@@ -78,10 +88,15 @@ void LRUFileCachePriority::pop(const CacheGuard::Lock &)
 
 LRUFileCachePriority::LRUQueueIterator LRUFileCachePriority::remove(LRUQueueIterator it)
 {
-    current_size -= it->size;
+    /// If size is 0, entry is invalidated, current_elements_num was already updated.
+    if (it->size)
+    {
+        current_size -= it->size;
+        --current_elements_num;
```

Comment:
Probably it'd be event better to have special methods (inlined) like `updateSize()` & `updateElementsCount()` which will update the value and corresponding metric, and update size/elements_count _only_ by using them